onnxruntime/core/session/environment.cc

```diff
@@ -216,14 +227,28 @@ Status Environment::Initialize(std::unique_ptr<logging::LoggingManager> logging_
     // Register Microsoft domain with min/max op_set version as 1/1.
     std::call_once(schemaRegistrationOnceFlag, []() {
       auto& domainToVersionRangeInstance = ONNX_NAMESPACE::OpSchemaRegistry::DomainToVersionRange::Instance();
+      if (domainToVersionRangeInstance.Map().find(onnxruntime::kMSDomain) == domainToVersionRangeInstance.Map().end()) {
+        // External shared providers may have already added kMSDomain
+        domainToVersionRangeInstance.AddDomainToVersion(onnxruntime::kMSDomain, 1, 1);
+      }
+      auto iter = domainToVersionRangeInstance.Map().find(onnxruntime::kOnnxDomain);
+      if (iter != domainToVersionRangeInstance.Map().end()) {
+        domainToVersionRangeInstance.AddDomainToVersion(onnxruntime::kMSInternalNHWCDomain, 1, iter->second.second);
+      } else {
```

Comment:
A small subset of operators are layout sensitive. If we go with this approach we should only be adding entries for that subset.